### PR TITLE
RF-57: Add Routine flow

### PR DESCRIPTION
**Pull request by AgustinCarthery**
# [RF-57: Add Routine flow](https://rutinesandfit.atlassian.net/jira/software/projects/RF/boards/1?selectedIssue=RF-57)

- Feature (new feature of the server/application)

## Description

- Creation of Routine stack with dashboard, add routine, select exercises and exercise revision screens
- useFormContext for routine creation flow
- added label prop on controlled text input and select

### CODE REVIEW

File: src/app/components/routine-badge/styles.ts [hunk 1]
```diff
@@ -5,10 +5,10 @@ export const StyledTouchableOpacity = styled.TouchableOpacity`
   border-radius: 8px;
   flex-direction: row;
   padding-horizontal: 25px;
-  padding-vertical: 20px;
+  padding-vertical: 18px;
   align-items: center;
   gap: 15px;
-  margin-bottom: 25px;
+  margin-bottom: 15px;
```
Comment (franco-marini) [inline src/app/components/routine-badge/styles.ts]: Could you use a value divisible of 4?

File: src/app/navigation/routines-stack.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+              onPress={() => {
+                navigation.goBack();
+              }}
+            />
+          }
+        />
+      ),
+    }),
+    [navigation, theme],
+  );
```
Comment (franco-marini) [inline src/app/navigation/routines-stack.tsx]: If you want something more performant than this, you could create this function outside the component and add the params navigation and theme. Then you call the function using the params.

WDYT?

Or can you review how we do it on the other stacks?

File: src/app/screens/routines/exercises-revision/components/exercise-item/index.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+} from 'src/app/components';
+import { AccordionItem } from 'src/app/components/accordion';
+
+import { ExerciseContainer, InputContainer, Row } from './styles';
+
+interface Props {
+  item: any;
+  control: any;
+  index: number;
+}
```
Comment (franco-marini) [inline src/app/screens/routines/exercises-revision/components/exercise-item/index.tsx]: Broooooo whyyyy you do this 😢

File: src/app/screens/routines/exercises-revision/components/exercise-item/index.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  Button,
+  ControlledSelect,
+  ControlledTextInput,
+  IconButton,
+} from 'src/app/components';
+import { AccordionItem } from 'src/app/components/accordion';
+
+import { ExerciseContainer, InputContainer, Row } from './styles';
+
+interface Props {
```
Comment (franco-marini) [inline src/app/screens/routines/exercises-revision/components/exercise-item/index.tsx]: Move all the types into a file and named the interfaces and types with the correct name, not just Props

File: src/app/screens/routines/form-config.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        },
+      ],
+    })),
+  };
+
+  return {
+    defaultValues,
+    resolver: yupResolver(validationSchema) as unknown as Resolver<FormData>,
+  };
+};
```
Comment (franco-marini) [inline src/app/screens/routines/form-config.ts]: I don't this is required to type this 🤔

**Review by franco-marini — COMMENTED** [review]
Nice work! Just a few comments to improve

File: src/app/screens/routines/form-config.ts (hunk 5 above)
Comment (AgustinCarthery) [reply src/app/screens/routines/form-config.ts]: what do you mean? previously this was typed: as Resolver<FormData> but encountered ts issues that led me to type as unknown also

**Review by franco-marini — APPROVED** [review]

**Pull request merged**